system.go

```diff
@@ -30,6 +30,21 @@ import (
 	_ "github.com/elastic/go-sysinfo/providers/windows"
 )
 
+type ProviderOption func(*registry.ProviderOptions)
+
+// WithHostFS returns a provider with a custom HostFS root path,
+// enabling use of the library from within a container, or an alternate root path on linux.
+// For example, WithHostFS("/hostfs") can be used when /hostfs points to the root filesystem of the container host.
+// For full functionality, the alternate hostfs should have:
+// - /proc
+// - /var
+// - /etc
```

Comment:
To get godoc rendering nicely

```suggestion
//   - /proc
//   - /var
//   - /etc
```